crates/viewer/re_space_view/src/annotation_context_utils.rs

```diff
@@ -0,0 +1,137 @@
+use ahash::HashMap;
+
+use re_types::components::Color;
+use re_viewer_context::{Annotations, QueryContext, ResolvedAnnotationInfos};
+
+use crate::clamped_or_nothing;
+
+#[inline]
+fn to_egui_color(color: &Color) -> egui::Color32 {
+    let [r, g, b, a] = color.to_array();
+    egui::Color32::from_rgba_unmultiplied(r, g, b, a)
+}
+
+/// Process [`Color`] components using annotations and default colors.
+pub fn process_color_slice<'a>(
+    ctx: &QueryContext<'_>,
+    fallback_provider: &'a dyn re_viewer_context::TypedComponentFallbackProvider<Color>,
+    num_instances: usize,
+    annotation_infos: &'a ResolvedAnnotationInfos,
+    colors: &'a [Color],
+) -> Vec<egui::Color32> {
+    // NOTE: Do not put tracing scopes here, this is called for every entity/timestamp in a frame.
```

Comment:
We could still have `puffin::profile_function_if(10_000 < num_instances);` to catch the cost of e.g. a big point cloud, without having to pay any overhead for small entities